ports/nrf/mphalport.c

```diff
@@ -35,6 +35,113 @@
 #include "nrfx_errors.h"
 #include "nrfx_config.h"
 
+#if MICROPY_PY_TIME_TICKS
+#include "nrfx_rtc.h"
+#include "nrf_clock.h"
+#endif
+
+
+#if MICROPY_PY_TIME_TICKS
+// use RTC1 for time ticks generation
+// with 32kHz tick resolution and overflow
+// handling in irq
+
+nrfx_rtc_t rtc1 = NRFX_RTC_INSTANCE(1);
+volatile mp_uint_t rtc_overflows = 0;
+
+const nrfx_rtc_config_t rtc_config_time_ticks = {
+    .prescaler = 0,
+    .reliable = 0,
+    .tick_latency = 0,
+    #ifdef NRF51
+    .interrupt_priority = 1,
+    #else
+    .interrupt_priority = 3,
+    #endif
+};
+
+STATIC void rtc_irq_time(nrfx_rtc_int_type_t event) {
+    // irq handler for overflow
+    if (event == NRFX_RTC_INT_OVERFLOW) {
+        rtc_overflows += 1;
+    }
+    // irq handler for wakeup from WFI (~1msec)
+    if (event == NRFX_RTC_INT_COMPARE0) {
+        nrfx_rtc_cc_set(&rtc1, 0, nrfx_rtc_counter_get(&rtc1) + 33, true);
+    }
+}
+
+void rtc1_init_time_ticks(void) {
+    // Start the low-frequency clock (if it hasn't been started already)
+    if (!nrf_clock_lf_is_running(NRF_CLOCK)) {
+        nrf_clock_task_trigger(NRF_CLOCK, NRF_CLOCK_TASK_LFCLKSTART);
+    }
+    nrfx_rtc_init(&rtc1, &rtc_config_time_ticks, rtc_irq_time);
```

Comment:
Should we add a `nrfx_rtc_counter_clear()` here? Right now, it remembers the old counter value across `soft reset`. Or is it more interesting to have the count from power on?

Reply:
I think just leave it, to keep it simple.  Ticks should anyway not be used as an absolute measure of time, just for relative differences (that's how it works in most other ports, it's only reset on power-up).

Reply:
I agree. I also noticed the documentation after writing the question which says: "... counter with an arbitrary reference point ...".

Reply:
> Should we add a nrfx_rtc_counter_clear() here?

see comment above